### DISCUSSION TITLE
I want to know if poco support windows arm?

### DISCUSSION THREAD

**Discussion by luohonghai** [pocoproject/poco#3762] [General]
I want to use this poco library in Hololens, is this support?

**Comment by aleks-f** [comment]
yes, @hmartinez82 is the maintainer of windows arm builds

**Reply by luohonghai** [reply]
thanks! @hmartinez82 can you help me how to build poco library used in Hololens?

**Reply by hmartinez82** [reply]
@luohonghai Sure. I could try 😊, but I don't have the hardware. Does it allow Win32 Subsystem apps or just WinRT/UWP ?

**Reply by luohonghai** [reply]
I am not sure it  allow Win32 Subsystem app, WinRT/UWP is sure

**Reply by hmartinez82** [reply]
It seems that Hololens 2 support native Win32. So what help do you need? :) Most of Poco already builds for ARM64 out of the box. Some components are not available for ARM64, for instance, the Data/MySQL can't be built today for ARM64 on Windows because there's no ARM64 SDK.. But OpenSSL does. https://docs.microsoft.com/en-us/windows/mixed-reality/develop/native/directx-development-overview?tabs=openxr

**Reply by luohonghai** [reply]
I want to use http and websocket function in UWP application, so I build PocoFoundation.dll with ARM64, it build successful, then link it in UWP application, when it run in Hololens, but it failed

**Reply by hmartinez82** [reply]
@luohonghai I don't have the hardware to help you test Hololens. If I were you I would try to compile an UWP for x64 first and try to use Poco there first to see if this is even possible.  This may not be a ARM64 problem, but rather a UWP problem.

aleks-f Are there reports of Poco running in UWP apps?

**Reply by luohonghai** [reply]
I think you are right, I have tried an UWP for x64,that's ok,  so I think maybe some windows library can use in x64 but not can use in ARM64,

**Reply by aleks-f** [reply]
> aleks-f Are there reports of Poco running in UWP apps?

I am not aware of it

**Reply by luohonghai** [reply]
Is there any plan to support Hololens hardware?thanks

**Reply by hmartinez82** [reply]
@luohonghai We should move this discussion to a new issue. Given the title of the issue there's nothing wrong today. The short answer is `yes`. Poco works for Windows on ARM. I've been using it in my Surface Pro X as a native application and not under UWP.

**Reply by hmartinez82** [reply]
If you want to file a new issue specific for Hololens 2, create a new issue and answer the following questions:
1. Are you trying to build Poco for Windows on ARM 64 outside UWP? Is it working for you?
2. Did you try building Poco for Windows x64 under UWP? Is that at least that working?
3. Hololens 2 accepts both UWP and Win32, Poco already works in the Win32 subsystem (for x86, x64 and ARM64)

You are asking if we are going to support Poco in Hololens, that's probably the wrong question. If what you are trying *must* be running under UWP then the right question would be if: Is there any plan to support Poco under UWP (regardless of the architecture) that's why you need to do step 2 above first.
